driver/controller.go

```diff
@@ -963,3 +973,46 @@ func validateCapabilities(caps []*csi.VolumeCapability) bool {
 
 	return supported
 }
+
+func (d *Driver) tagVolume(ctx context.Context, vol *godo.Volume) error {
+	for _, tag := range vol.Tags {
+		if tag == d.doVolTag {
+			return nil
+		}
+	}
+
+	tagReq := &godo.TagResourcesRequest{
+		Resources: []godo.Resource{
+			godo.Resource{
+				ID:   vol.ID,
+				Type: godo.VolumeResourceType,
+			},
+		},
+	}
+
+	ctx, cancel := context.WithTimeout(ctx, doAPITimeout)
+	resp, err := d.tags.TagResources(ctx, d.doVolTag, tagReq)
+	cancel()
+
+	if resp != nil && resp.StatusCode == http.StatusNotFound {
+		// If the tag doesn't exist, create it and retry.
+		err = d.createTag(ctx)
+		if err != nil {
+			return err
+		}
+		ctx, cancel = context.WithTimeout(ctx, doAPITimeout)
+		resp, err = d.tags.TagResources(ctx, d.doVolTag, tagReq)
+		cancel()
+	}
+
+	return err
+}
+
+func (d *Driver) createTag(ctx context.Context) error {
+	ctx, cancel := context.WithTimeout(ctx, doAPITimeout)
+	defer cancel()
+	_, _, err := d.tags.Create(ctx, &godo.TagCreateRequest{
+		Name: d.doVolTag,
+	})
+	return err
+}
```

Comment:
Is there a reason we're wrapping this call here? We're using it in only one place and I don't think we should create a function for this.

Reply:
I originally did this to try to cleanup some of the multiple context mess in `tagVolume`.  You're right to suggest pulling this into that function.  Done.